x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsServiceSettings.java

```diff
@@ -34,62 +37,108 @@
 import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
 
 /**
- * Defines the base settings for interacting with OpenAI.
+ * Defines the service settings for interacting with OpenAI's text embedding models.
  */
-public class OpenAiServiceSettings implements ServiceSettings {
+public class OpenAiEmbeddingsServiceSettings implements ServiceSettings {
 
     public static final String NAME = "openai_service_settings";
 
     public static final String ORGANIZATION = "organization_id";
+    static final String DIMENSIONS_SET_BY_USER = "dimensions_set_by_user";
```

Comment:
Good point, I'll switch `ORGANIZATION` to be package private. `NAME` needs to be public because it's used in a separate package when registering the named writeable registry.